app/src/main/java/com/celzero/bravedns/util/Constants.kt

```diff
@@ -15,158 +15,182 @@ limitations under the License.
         const val SOCKS = "Socks5"
 
+        // data-time format used as part of network log adapter
         const val DATE_FORMAT_PATTERN = "HH:mm:ss"
 
+        // represents the orbot proxy mode (see AppMode#ProxyMode)
         const val ORBOT_PROXY = 10L
 
-        // Represents the download source of the application. playstore/fdroid/website
-        const val DOWNLOAD_SOURCE_PLAY_STORE = 1
-        const val DOWNLOAD_SOURCE_FDROID = 2
-        const val DOWNLOAD_SOURCE_WEBSITE = 3
-
-        // Constants generated as part of BuildConfig.FLAVORS
+        // constants generated as part of BuildConfig.FLAVORS (playstore/fdroid/website)
         const val FLAVOR_PLAY = "play"
         const val FLAVOR_FDROID = "fdroid"
         const val FLAVOR_WEBSITE = "website"
 
+        // represents the unknown port in the port map. see class KnownPorts
         const val PORT_VAL_UNKNOWN = "unknown"
 
         // Various notification action constants used part of NotificationCompat.Action
         const val NOTIFICATION_ACTION = "NOTIFICATION_VALUE"
-        const val NOTIF_ACTION_STOP_VPN = "RETHINK_STOP"
+        const val NOTIF_ACTION_STOP_VPN = "RETHINK_STOP" // stop vpn service
+        const val NOTIF_ACTION_PAUSE_VPN = "RETHINK_PAUSE" // pause vpn
+        const val NOTIF_ACTION_RESUME_VPN = "RETHINK_RESUME" // resume vpn
         const val NOTIF_ACTION_DNS_VPN = "RETHINK_DNSONLY" // battery-saver dns-only
         const val NOTIF_ACTION_DNS_FIREWALL_VPN = "RETHINK_FULLMODE" // default dns+firewall
         const val NOTIF_ACTION_RULES_FAILURE = "RETHINK_RULES_RELOAD" // load rules failure
+        const val NOTIF_ACTION_NEW_APP_ALLOW = "NEW_APP_ALLOW" // allow network access for new apps
+        const val NOTIF_ACTION_NEW_APP_DENY = "NEW_APP_DENY" // deny network access for new apps
 
+        // various notification intent extra name/values used part of notification's pending-intent
+        const val NOTIF_INTENT_EXTRA_ACCESSIBILITY_NAME = "ACCESSIBILITY" // accessibility failure name
+        const val NOTIF_INTENT_EXTRA_ACCESSIBILITY_VALUE = "ACCESSIBILITY_FAILURE" // accessibility failure value
+        const val NOTIF_INTENT_EXTRA_NEW_APP_NAME = "NEW_APP" // new app install name
+        const val NOTIF_INTENT_EXTRA_NEW_APP_VALUE = "NEW_APP_INSTALL_NOTIFY" // new app install value
 
-        const val NOTIF_INTENT_EXTRA_ACCESSIBILITY_NAME = "ACCESSIBILITY_GRANT"
-        const val NOTIF_INTENT_EXTRA_ACCESSIBILITY_VALUE = "ACCESSIBILITY_FAILURE"
+        // new app install intent extra name for uid. see RefreshDatabase#makeNewAppVpnIntent()
+        const val NOTIF_INTENT_EXTRA_APP_UID = "NEW_APP_UID"
 
+        // DNS message type received by the DNS resolver
         const val NXDOMAIN = "NXDOMAIN"
 
         // Maximum time out for the DownloadManager to wait for download of local blocklist.
         // The time out value is set as 40 minutes.
         val WORK_MANAGER_TIMEOUT = TimeUnit.MINUTES.toMillis(40)
```

Comment:
move to downloadmanager?

rename `ONDEVICE_BLOCKLIST_DOWNLOAD_TIMEOUT_MS`

Reply:
Moved it to `AppDownloadManager`